src/core/network.js

```diff
@@ -15,88 +15,89 @@ class Network extends EventHandler {
         super();
     }
 
-    initialize(levelProvider, onBeforeConnection) {
-        if (this.io)
+    initialize(levelProvider) {
+        if (this.server)
             return;
 
-        global.LevelProvider = levelProvider;
-
-        // const options = {
-        //     key: readFileSync('privkey.pem', 'utf8'),
-        //     cert: readFileSync('cert.pem', 'utf8'),
-        //     ca: readFileSync('chain.pem', 'utf8'),
-        // };
-
-        const httpServer = createServer();
-        this.io = new Server(httpServer, {
-            cors: {
-                origin: '*',
-                methods: ['GET', 'POST'],
-            }
-        });
-        httpServer.listen(this.port);
-
-        if (onBeforeConnection)
-            this.io.use(onBeforeConnection);
-
-        this.io.on('connection', (socket) => {
-            const user = socket.user = new User(socket, socket.id, socket.data);
-
-            socket.on('disconnecting', () => {
-                user.rooms.forEach((room) => {
-                    room.leave(user);
-                });
-
-                this.fire('user:disconnected', user);
-            });
-
-            // send basic information to connected user
-            user.send('self', {
-                userId: user.id,
-                data: user.data,
-                templates: templates.toData()
-            });
-
-            socket.on('room:create', async (levelId, roomType, callback) => {
+        const handlers = {
```

Comment:
Create Issue, to remove handlers, and move room:* to an example logic, not core.